pkg/proxy/iptables/proxier.go

```diff
@@ -975,6 +975,16 @@ func (proxier *Proxier) syncProxyRules() {
 	if err != nil {
 		klog.ErrorS(err, "Failed to get node ip address matching nodeport cidrs, services with nodeport may not work as intended", "CIDRs", proxier.nodePortAddresses)
 	}
+	// nodeAddresses may contain dual-stack zero-CIDRs if proxier.nodePortAddresses is empty.
```

Comment:
I see `GetNodeAddresses` returns 2 zero addresses

https://github.com/kubernetes/kubernetes/blob/6ce612ef65c6ac67a7f67a1c12eec970c05a6c6f/pkg/proxy/util/utils.go#L199-L205

can we have the same issue with dual stack services?

Reply:
For each Proxier, the patch removes the other zero address from `nodeAddresses`, so the two Proxiers can open their own ports without conflict. I have tested it:
```
tcp        0      0 0.0.0.0:31915           0.0.0.0:*               LISTEN      1756224/kube-proxy
tcp6       0      0 :::31915                :::*                    LISTEN      1756224/kube-proxy
```

Reply:
awesome